src/Raven.Server/Documents/Queries/Parser/QueryParser.cs

```diff
@@ -91,12 +91,46 @@ public Query Parse(QueryType queryType = QueryType.Select)
                     break;
             }
 
+            Paging(out q.Offset, out q.Limit);
+
             if (Scanner.AtEndOfInput() == false)
                 ThrowParseException("Expected end of query");
 
             return q;
         }
 
+        private void Paging(out ValueExpression offset, out ValueExpression limit)
+        {
+            offset = null;
+            limit = null;
+
+            if (Scanner.TryScan("LIMIT"))
+            {
+                if (Value(out var first) == false)
+                    throw new InvalidQueryException("Limit must contain a value", Scanner.Input, null);
+
+                if (Scanner.TryScan(","))
+                {
+                    if (Value(out var second) == false)
+                        throw new InvalidQueryException("Limit must contain a second value", Scanner.Input, null);
+
+                    offset = first;
+                    limit = second;
+                }
```

Comment:
If there is a second item there, need to exit here.
Otherwise, you have `limit 10, 2 offset 3`, which doesn't make sense.

Reply:
I did it because I'm detecting if 'offset' is already set to throw a nice exception to the user

Reply:
okay